### DISCUSSION TITLE
Blank Interface

### DISCUSSION THREAD

**Discussion by dtdionne** [modoboa/modoboa#3228] [Help] [answered]
Bone stock install with quick installer on Debian 12.  Admin interface is blank.  Going to https://ip or localhost results in a 400 bad request.

Any ideas?

**Comment by dtdionne** [comment]
Well theres no nginx config.  How in the hell is this coined an "easy install"?

You people do all this amazing work and then poop the bed in the last inch of the last foot in a marathon.  It makes no sense.  If an "easy install" aint that ur just shooting urself in the foot.

There's nothing in the install doc's that talks about having to fully configure nginx after the "easy install"...how much else isn't configured?

**Comment by dtdionne** [comment]
HAAAHAHAHAHA!!!  And none of the answers are in the docs!  It just says "do it" "use the modoaba_instance_path"!!!  This is actually now quite entertaining.

This is a treat!

https://modoboa.readthedocs.io/en/2.2.4/manual_installation/webserver.html

**Comment by Toniob** [comment]
Do you really think that’s a way to ask for help ? Do you really think that’s a way to speak to people ? People who are giving this piece of software for free, doing it on their spare time. Do you think you’re going to help anyone with your behavior ?

**Reply by dtdionne** [reply]
I'm sorry bud, im an old school IT expert.  Cut my teeth at motorola in early 90's.  We had hundreds of toll free lines jam packed 24/7 for almost a year becuase someone screwed up a huge ISDN shipment.  You wouldn't believe the profanity and threats i got everyday, all day, for a year.  Didnt bother me one bit.  I knew i could help and i couldnt wait to do just that.  Again, im sorry bud, this hypersensitivity makes no sense to me.  And if we stay on the trajectory we'll all be sealed in  medically induced coma's, in goo filled drums for safety for fear the slightest sound or movement will break someone's heart.

Lets try and toughen up a schootch maybe?  It'll do ya good.

**Reply by dtdionne** [reply]
Now i'd like to try modoboa but there are many other options...

**Reply by dtdionne** [reply]
So did my install go sideways, or is this "how it is"?

david@Modoboa:~/modoboa-installer$ sudo python3 run.py 'sanitized'
Welcome to Modoboa installer!

Configuration file installer.cfg not found, creating new one.
Warning:
Before you start the installation, please make sure the following DNS records exist for domain 'sanitized':
  mail IN A   <IP ADDRESS OF YOUR SERVER>
     @ IN MX  'sanitized'.

Your mail server will be installed with the following components:
fail2ban modoboa automx amavis clamav dovecot nginx razor postfix postwhite spamassassin uwsgi radicale opendkim backup
Do you confirm? (Y/n) y
The process can be long, feel free to take a coffee and come back later ;)
Starting...
Generating new self-signed certificate
Installing amavis
Installing spamassassin
Installing razor
Installing clamav
Installing fail2ban
Installing modoboa
Installing automx
Installing radicale
Installing uwsgi
Installing nginx
Installing opendkim
Installing postfix
Installing postwhite
Installing dovecot
User dovecot already exists, skipping creation but please make sure the /srv/vmail directory exists.
Congratulations! You can enjoy Modoboa at https://'sanitized'

**Reply by Toniob** [reply]
Don’t start your answer with "I’m sorry" when you obviously aren’t sorry at all. And I clearly see how you are. Trying to tell it’s my fault, that I should be more tough, instead of looking at your behavior. You treat people like shit, and when they tell you what an horrible person you are, you’re searching for bad reasons to justify that. But the only guilty person is yourself. I hope that one day you’ll understand and you’ll stop blaming someone else for your shitty behavior.

You’re an IT expert ? You’re probably way better than me, as I don’t describe myself as one. Oh, by the way, I installed Modoboa several times, without the installer, on my own. So you’ll probably find how to do it yourself too. Show us how good an expert you are, douchebag !

**Reply by dtdionne** [reply]
I am sorry it hurt feelings, but they are your feelings and therefore your responsibility.

**Comment by dtdionne** [comment, marked as answer]
I got it, it was the TZ setting in /srv/modoboa/instance/instance/settings.py.  Both...

1. disabling TZ
2. correcting TZ ('America/Chicago' in my case)

Fixed the issue.

That seems like a painful and silly bug that unjustly shines Modoboa in unfavorable light.

**Comment by damienvancouver** [comment]
> I'm sorry bud, im an old school IT expert. Cut my teeth at motorola in early 90's. 

ohhhh cool story, Bro!!!!!

> Lets try and toughen up a schootch maybe? It'll do ya good.

You should try and be more constructive when mooching support, it'll do you more good!  Or at least get you better results!


FWIW I just did a clean Debian 12 install, none of your issues occurred for me...   Modoboa settings.py contains a default Timezone (Europe/Paris) afterwards, nginx loaded the admin interface just fine and everything is working except the Calendar.

What was your TZ set to originally?  Was it a clean/empty Debian 12 or an existing one with existing baggage?  The installer does not always run so cleanly on an upgrade or on an existing machine with other stuff going on.  But it does work fine from a clean VM, to so it would appear your install did go sideways for whatever reason (plenty of reasons that could be).

What TZ was in there before?    What was the error message?  In future you should try setting `DEBUG = True` in settings.py, then restarting uwsgi, then errors and stack traces are displayed nicely in the browser.

**Reply by dtdionne** [reply]
It was a brand new Deb 12 downloaded and installed this morning.  There wasn’t an error message, the browser was just blank.  I saw the error in a full-frame hex/ascii tcpdump.

TZ was set to Paris, while the system is and was set to US/Central during the install.  And I was wrong, it didn’t have anything to do with nginx…it was just that TZ setting.  Once that was fixed everything else seems to work fine.

I can run it again, it’s a vm and I took a snapshot of the pristine deb install.  I’ll spinoff a clone and run the debug installer tomorrow, see if it happens again.

**Reply by dtdionne** [reply]
Now that I’ve seen it I’ll probably install on centos too, being an RHCE (and an MCSE) I’m more familiar with rhel otb.

**Reply by dtdionne** [reply]
Until my ARIN IPv4 transfer comes through, new ASN  and bgp config I only have a single public ip.  I was planning on putting nginx out front to proxy 443 on host headers.  Anyone had issues trying this with modoboa?  I can’t imagine there would be but can’t hurt to ask.

**Reply by dtdionne** [reply]
Never mind, it works great.  So until BGP here’s to nginx.

**Reply by dtdionne** [reply]
I just did a fresh install on Ubuntu with certbot (letsEncrypt) and the way the installer processes certbot it doesn’t create the live dir in /etc/letsencrypt.  The certs are there but in a csr dir so nginx.conf needs some massaging to get it running.  I’ve verified this on deb and ununtu.

Even massaged those certs didn’t work.  I have wildcards so I only spent a couple min on it before switching to those.  I didn’t have the TZ issue on Ubuntu server.

**Reply by dtdionne** [reply]
I’m not gonna lie, this is impressive work…well done!

It took a little finagling but hosting email has always been a bramble bushed, quicksanded, tar pit.  I’ll add this next part as a follow-on answer…

The letsEncrypt hiccup I was having was 100% user error.  It was caused by the nginx proxy i put out front.  The default “server” served the stock nginx web page from the proxy which certbot didn’t like.  So adding an additional http server location proxying to the actual modoboa server fixed that during install.  And the certbot certificates ripple through the whole install and app.  I didn’t look real close but by scanning the debug log as it installs I spotted a couple places they’re used.

I think I’ve settled on Ubuntu server as the host OS, it’s enough debian’ish that the install cooks along fine and enough rhel’ish that I’m cool with that.  At least it still has netstat and ifconfig available.  You youngsters will know what I mean in 30 years when so many of the commands you’ve used for 30 years get replaced for whatever reason.  Those commands and all the switches become both physical and mental muscle memory.  And it doesn’t take many being replaced to make u feel a little uncomfortable.

**Comment by dtdionne** [comment]
More regarding the certbot (letsEncrypt) issue I had which was 100% user (my) error.  And real certificates are a big deal.  This is much more of a corner case than a mainline issue but…

It was caused by the nginx proxy i put out front.  The default http (port 80 which is used by certbot verification) “server” served the stock nginx web page from the proxy server and not the modoboa server which certbot wasn’t thrilled with.  So adding an additional http server location proxying to the actual modoboa server fixed that during install.  And the certbot certificates ripple through the whole install and app.  I didn’t look real close but by scanning the debug log as it installs I spotted a couple places they’re used.  It might be worthwhile to add an interactive choice for real certs during install and pause the install if there’s a hiccup.

The TZ hiccup didn’t happen on Ubuntu server.  I haven’t gotten around to trying on Debian again yet.
